plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/transport/reload/ReloadPipelinesAction.java

```diff
@@ -0,0 +1,123 @@
+
+import org.elasticsearch.cluster.ClusterService;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.component.AbstractComponent;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.plugin.ingest.PipelineStore;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+/**
+ * An internal api that refreshes the in-memory representation of all the pipelines on all ingest nodes.
+ */
+public class ReloadPipelinesAction extends AbstractComponent implements TransportRequestHandler<ReloadPipelinesAction.ReloadPipelinesRequest> {
+
+    public static final String ACTION_NAME = "internal:admin/ingest/reload/pipelines";
+
+    private final ClusterService clusterService;
+    private final TransportService transportService;
+    private final PipelineStore pipelineStore;
+
+    public ReloadPipelinesAction(Settings settings, PipelineStore pipelineStore, ClusterService clusterService, TransportService transportService) {
+        super(settings);
+        this.pipelineStore = pipelineStore;
+        this.clusterService = clusterService;
+        this.transportService = transportService;
+        transportService.registerRequestHandler(ACTION_NAME, ReloadPipelinesRequest::new, ThreadPool.Names.SAME, this);
+    }
+
+    public void reloadPipelinesOnAllNodes(Consumer<Boolean> listener) {
+        List<DiscoveryNode> ingestNodes = new ArrayList<>();
+        for (DiscoveryNode node : clusterService.state().getNodes()) {
+            String nodeEnabled = node.getAttributes().get("ingest");
+            if ("true".equals(nodeEnabled)) {
+                ingestNodes.add(node);
+            }
+        }
+
+        if (ingestNodes.isEmpty()) {
+            throw new IllegalStateException("There are no ingest nodes in this cluster");
+        }
+
+        AtomicBoolean failed = new AtomicBoolean();
+        AtomicInteger expectedResponses = new AtomicInteger(ingestNodes.size());
+        for (DiscoveryNode node : ingestNodes) {
+            ReloadPipelinesRequest nodeRequest = new ReloadPipelinesRequest();
+            transportService.sendRequest(node, ACTION_NAME, nodeRequest, new TransportResponseHandler<ReloadPipelinesResponse>() {
+                @Override
+                public ReloadPipelinesResponse newInstance() {
+                    return new ReloadPipelinesResponse();
+                }
+
+                @Override
+                public void handleResponse(ReloadPipelinesResponse response) {
+                    decrementAndReturn();
+                }
+
+                @Override
+                public void handleException(TransportException exp) {
+                    logger.warn("failed to update pipelines on remote node [{}]", exp, node);
```

Comment:
It would be great to add a test for this scenario.

An ingest node may have disconnected or randomly stopped being an ingest node between the time that 
`node.getAttributes().get("ingest")` was called, and when the `ReloadPipelineRequest` was actually issued against it.